From a5727e0499f42834c77a2caad78005ab9da1534f Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho.andersen@canonical.com>
Date: Thu, 7 Jan 2016 15:17:38 -0700
Subject: bash completion updates

* rename to lxd-client to avoid conflicting with lxc's completer
* add a few missing commands and completions
* guard the whole thing with `have lxc`, as is done in other packages

Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>
---
 config/bash/lxc.in     | 80 ------------------------------------------
 config/bash/lxd-client | 95 ++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 95 insertions(+), 80 deletions(-)
 delete mode 100644 config/bash/lxc.in
 create mode 100644 config/bash/lxd-client

diff --git a/config/bash/lxc.in b/config/bash/lxc.in
deleted file mode 100644
index 9664b30..0000000
--- a/config/bash/lxc.in
+++ /dev/null
@@ -1,80 +0,0 @@
-_lxc_complete()
-{
-  _lxc_names()
-  {
-    COMPREPLY=( $( compgen -W \
-      "$( lxc list | tail -n +4 | awk '{print $2}' | egrep -v '^(\||^$)' )" "$cur" )
-    )
-  }
-
-  _lxc_images()
-  {
-    COMPREPLY=( $( compgen -W \
-      "$( lxc image list | tail -n +4 | awk '{print $2}' | egrep -v '^(\||^$)' )" "$cur" )
-    )
-  }
-
-  local cur prev
-
-  COMPREPLY=()
-  cur=${COMP_WORDS[COMP_CWORD]}
-  prev=${COMP_WORDS[COMP_CWORD-1]}
-  lxc_cmds="config copy delete exec file finger help image info init launch \
-    list move profile remote restart restore snapshot start stop version"
-
-  if [ $COMP_CWORD -eq 1 ]; then
-    COMPREPLY=( $(compgen -W "$lxc_cmds" -- $cur) )
-  elif [ $COMP_CWORD -eq 2 ]; then
-    case "$prev" in
-      "config")
-        COMPREPLY=( $(compgen -W "device edit get set show trust" -- $cur) )
-        ;;
-      "exec")
-        _lxc_names
-        ;;
-      "exec")
-        COMPREPLY=( $(compgen -W "pull push" -- $cur) )
-        ;;
-      "help")
-        COMPREPLY=( $(compgen -W "$lxc_cmds" -- $cur) )
-        ;;
-      "image")
-        COMPREPLY=( $(compgen -W "copy delete edit export info list" -- $cur) )
-        ;;
-      "info")
-        _lxc_names
-        ;;
-      "launch")
-        _lxc_images
-        ;;
-      "profile")
-        COMPREPLY=( $(compgen -W \
-          "list show create edit copy set delete apply" -- $cur) )
-        ;;
-      "remote")
-        COMPREPLY=( $(compgen -W \
-          "add remove list rename set-url set-default get-default" -- $cur) )
-        ;;
-      "restart")
-        _lxc_names
-        ;;
-      "snapshot")
-        _lxc_names
-        ;;
-      "start")
-        # should check if containers are stopped
-        _lxc_names
-        ;;
-      "stop")
-        # should check if containers are started
-        _lxc_names
-        ;;
-      *)
-        ;;
-    esac
-  fi
-
-  return 0
-}
-
-complete -F _lxc_complete lxc
diff --git a/config/bash/lxd-client b/config/bash/lxd-client
new file mode 100644
index 0000000..90eda94
--- /dev/null
+++ b/config/bash/lxd-client
@@ -0,0 +1,95 @@
+have lxc && {
+  _lxd_complete()
+  {
+    _lxd_names()
+    {
+      COMPREPLY=( $( compgen -W \
+        "$( lxc list | tail -n +4 | awk '{print $2}' | egrep -v '^(\||^$)' )" "$cur" )
+      )
+    }
+
+    _lxd_images()
+    {
+      COMPREPLY=( $( compgen -W \
+        "$( lxc image list | tail -n +4 | awk '{print $2}' | egrep -v '^(\||^$)' )" "$cur" )
+      )
+    }
+
+    local cur prev
+
+    COMPREPLY=()
+    cur=${COMP_WORDS[COMP_CWORD]}
+    prev=${COMP_WORDS[COMP_CWORD-1]}
+    lxc_cmds="config copy delete exec file help image info init launch \
+      list move profile publish remote restart restore snapshot start stop \
+      version"
+
+    if [ $COMP_CWORD -eq 1 ]; then
+      COMPREPLY=( $(compgen -W "$lxc_cmds" -- $cur) )
+    elif [ $COMP_CWORD -eq 2 ]; then
+      case "$prev" in
+        "config")
+          COMPREPLY=( $(compgen -W "device edit get set show trust" -- $cur) )
+          ;;
+        "copy")
+          _lxd_names
+          ;;
+        "delete")
+          _lxd_names
+          ;;
+        "exec")
+          _lxd_names
+          ;;
+        "exec")
+          COMPREPLY=( $(compgen -W "pull push" -- $cur) )
+          ;;
+        "help")
+          COMPREPLY=( $(compgen -W "$lxc_cmds" -- $cur) )
+          ;;
+        "image")
+          COMPREPLY=( $(compgen -W "copy delete edit export info list" -- $cur) )
+          ;;
+        "info")
+          _lxd_names
+          ;;
+        "launch")
+          _lxd_images
+          ;;
+        "profile")
+          COMPREPLY=( $(compgen -W \
+            "list show create edit copy set delete apply" -- $cur) )
+          ;;
+        "publish")
+          _lxd_names
+          ;;
+        "remote")
+          COMPREPLY=( $(compgen -W \
+            "add remove list rename set-url set-default get-default" -- $cur) )
+          ;;
+        "restart")
+          _lxd_names
+          ;;
+        "restore")
+          _lxd_names
+          ;;
+        "snapshot")
+          _lxd_names
+          ;;
+        "start")
+          # should check if containers are stopped
+          _lxd_names
+          ;;
+        "stop")
+          # should check if containers are started
+          _lxd_names
+          ;;
+        *)
+          ;;
+      esac
+    fi
+
+    return 0
+  }
+
+  complete -F _lxd_complete lxc
+}
